File: advertising_sdk/src/main/java/com/adadapted/android/sdk/core/atl/PopupClient.java
package com.adadapted.android.sdk.core.atl;

import com.adadapted.android.sdk.core.concurrency.ThreadPoolInteractorExecuter;
import com.adadapted.android.sdk.core.event.AppEventClient;

import java.util.HashMap;
import java.util.Map;

class PopupClient {
    static void markPopupContentAcknowledged(final PopupContent content) {
        ThreadPoolInteractorExecuter.getInstance().executeInBackground(new Runnable() {
            @Override
            public void run() {
                final Map<String, String> params = new HashMap<>();
                params.put("payload_id", content.getPayloadId());

                AppEventClient.trackSdkEvent("popup_added_to_list", params);
            }
        });
    }

    static void markPopupContentItemAcknowledged(final PopupContent content, final AddToListItem item) {
        ThreadPoolInteractorExecuter.getInstance().executeInBackground(new Runnable() {
            @Override
            public void run() {
                final Map<String, String> params = new HashMap<>();
                params.put("payload_id", content.getPayloadId());
                params.put("tracking_id", item.getTrackingId());
                params.put("item_name", item.getTitle());

                AppEventClient.trackSdkEvent("popup_item_added_to_list", params);
            }
        });
    }

    static void markPopupContentFailed(final PopupContent content, final String message) {
        ThreadPoolInteractorExecuter.getInstance().executeInBackground(new Runnable() {
            @Override
            public void run() {
                final Map<String, String> eventParams = new HashMap<>();
                eventParams.put("payload_id", content.getPayloadId());

                AppEventClient.trackError("POPUP_CONTENT_FAILED", message, eventParams);
            }
        });
    }

    static void markPopupContentItemFailed(final PopupContent content, final AddToListItem item, final String message) {
        ThreadPoolInteractorExecuter.getInstance().executeInBackground(new Runnable() {
            @Override
            public void run() {
                final Map<String, String> eventParams = new HashMap<>();
                eventParams.put("payload_id", content.getPayloadId());
                eventParams.put("tracking_id", item.getTrackingId());

                AppEventClient.trackError("POPUP_CONTENT_ITEM_FAILED", message, eventParams);
            }
        });
    }
}

File: advertising_sdk/src/main/java/com/adadapted/android/sdk/core/atl/PopupClient.kt
package com.adadapted.android.sdk.core.atl

import com.adadapted.android.sdk.config.EventStrings
import com.adadapted.android.sdk.core.event.AppEventClient.Companion.getInstance

internal object PopupClient {
    fun markPopupContentAcknowledged(content: PopupContent) {
        val params: MutableMap<String, String> = HashMap()
        params[PAYLOAD_ID] = content.payloadId
        getInstance().trackSdkEvent(EventStrings.POPUP_ADDED_TO_LIST, params)
    }

    fun markPopupContentItemAcknowledged(content: PopupContent, item: AddToListItem) {
        val params: MutableMap<String, String> = HashMap()
        params[PAYLOAD_ID] = content.payloadId
        params[TRACKING_ID] = item.trackingId
        params[ITEM_NAME] = item.title
        getInstance().trackSdkEvent(EventStrings.POPUP_ITEM_ADDED_TO_LIST, params)
    }

    fun markPopupContentFailed(content: PopupContent, message: String) {
        val eventParams: MutableMap<String, String> = HashMap()
        eventParams[PAYLOAD_ID] = content.payloadId
        getInstance().trackError(EventStrings.POPUP_CONTENT_FAILED, message, eventParams)
    }

    fun markPopupContentItemFailed(content: PopupContent, item: AddToListItem, message: String) {
        val eventParams: MutableMap<String, String> = HashMap()
        eventParams[PAYLOAD_ID] = content.payloadId
        eventParams[TRACKING_ID] = item.trackingId
        getInstance().trackError(EventStrings.POPUP_CONTENT_ITEM_FAILED, message, eventParams)
    }

    private const val PAYLOAD_ID = "payload_id"
    private const val TRACKING_ID = "tracking_id"
    private const val ITEM_NAME = "item_name"
}